update video uploads page

muhammad-ammar opened the pull request:
[EDUCATOR-1037](https://openedx.atlassian.net/browse/EDUCATOR-1037)

[SANDBOX](https://studio-update-video-uploads.sandbox.edx.org/videos/course-v1:edX+DemoX+Demo_Course)

muhammad-ammar commented:
@muzaffaryousaf This is ready for review.

srpearce commented on cms/templates/js/active-video-upload-list.underscore:
```diff
@@ -1,6 +1,25 @@
+        <div class="video-allowed-extensions-text"><%-
+            edx.StringUtils.interpolate(
+                // Translators: Extensions of videos allowed to upload
+                gettext('Extensions allowed: {videoExtensions}'),
```
Supported file types: {videoExtensions}

srpearce commented on cms/static/js/spec/views/active_video_upload_list_spec.js:
```diff
@@ -57,6 +57,23 @@ define(
+                            'Max file size: {maxFileSize} GB', {maxFileSize: this.videoUploadMaxFileSizeInGB}
+                        ),
+                        '.video-allowed-extensions-text': StringUtils.interpolate(
+                            'Extensions allowed: {extensions}', {extensions: this.videoSupportedFileFormats.join(', ')}
```
Supported file types: {videoExtensions}

srpearce commented on cms/static/js/spec/views/active_video_upload_list_spec.js:
```diff
@@ -57,6 +57,23 @@ define(
+                var messages = {
+                        '.video-upload-text': 'Drag and drop or click here to upload video files.',
+                        '.video-max-file-size-text': StringUtils.interpolate(
+                            'Max file size: {maxFileSize} GB', {maxFileSize: this.videoUploadMaxFileSizeInGB}
```
Max --> Maximum

srpearce commented on cms/static/js/spec/views/active_video_upload_list_spec.js:
```diff
@@ -57,6 +57,23 @@ define(
 
+            it('renders correct text in file drag/drop area', function() {
+                var messages = {
+                        '.video-upload-text': 'Drag and drop or click here to upload video files.',
```
Drag and drop your video files here, or click this space to browse for files on your computer.

muhammad-ammar commented:
jenkins run js

mushtaqak commented on cms/static/js/spec/views/active_video_upload_list_spec.js:
```diff
@@ -57,6 +57,23 @@ define(
+                    },
+                    self = this;
+
+                Object.keys(messages).forEach(function(messageClass) {
```
I hope `Object.keys` is supported by all browsers that edX supports, right ?

muhammad-ammar replied:
We are using `Object.keys` only in the test, So there will no issue.

roderickmorales commented:
![video-upload-page](https://user-images.githubusercontent.com/1260455/28776029-c7aa8244-75c2-11e7-875b-9ec08d225714.png)

Add a header for “Upload Videos”
Add an upload icon
Center align the div containing icon and text
Left align text
“Browse your computer” to look and feel like a link
Add the total upload size 


fa-cloud-upload{
font-size:7em;
float: left;
marian-right: 0.25em
}

“Upload Videos”{
font-size: 1.5em;
margin-bottom: .25em;
font-weight: 600
}

REMOVE
.view-video-uploads .file-upload-form{
	margin-bottom: 30px;
}

UPDATE TO
.view-video-uploads .file-upload-form .file-drop-area {
	padding: 25px;
	text-align: left;
}

muhammad-ammar commented:
@roderickmorales Thanks for the css styles. 
One question, what exactly is `(XXXGB max total upload)`?

muhammad-ammar commented:
@srpearce Updated text strings as you suggested. Can we make `Drag and drop your video files here, or click this space to browse for files on your computer.` shorter? One option could be `Drag and drop videos here, or click this space to browse videos on your computer.` OR even better if we can make string length same to the original one?

muhammad-ammar commented:
@roderickmorales Addressed your feedback. The whole drag and drop area is clickable so I preferred not to make an anchor in the text. But if you insist than we can create an anchor in some future PR.

roderickmorales commented:
@muhammad-ammar Is not a personal opinion to have it "look" like a link. It is a best UX practice, without looking like a link then it is really hard to quickly scan the page and realize that you can upload files without dragging and dropping. 

Look at how GitHub does it when you are adding a comment (screenshot attached). If you inspect the HTML you realize that it is not a link but it feels like a link there for boosting the affordance of this feature. We should be able to do this pretty easily.

<img width="701" alt="screen shot 2017-08-01 at 9 07 28 am" src="https://user-images.githubusercontent.com/1260455/28826659-df7acaf6-7698-11e7-9b9a-e633f58e7ce7.png">

FYI @muzaffaryousaf

srpearce commented:
@muhammad-ammar @mushtaqak How about "Drag and drop videos here, or click this space to browse your computer"?

muhammad-ammar commented:
Thanks @roderickmorales @srpearce I will address your feedback.

sstack22 commented:
This looks great, 👍 !

muzaffaryousaf commented:
@muhammad-ammar can you add styling for this and let's get rid of this PR ?

muhammad-ammar commented:
> can you add styling for this and let's get rid of this PR ?

@muzaffaryousaf will finish it today.

muhammad-ammar commented:
@srpearce @roderickmorales Addressed the feedback and sandbox is updated.

mushtaqak commented on cms/static/js/views/active_video_upload_list.js:
```diff
@@ -23,10 +23,18 @@ define([
 
+            uploadHeader: gettext('Upload Videos'),
+            uploadText: HtmlUtils.interpolateHtml(
+                gettext('Drag and drop videos here, or click this space to {spanStart}browse your computer{spanEnd}.'),
```
My suggestion is `Drag and drop videos here, or click here to browse your computer`

roderickmorales commented:
@srpearce @mushtaqak looks good but will really prefer not to use the word "click." Think of a user that uses a keyboard or a touch screen. The only time you actually "click" on something is when you use a mouse.

srpearce commented:
@roderickmorales Do you have a suggestion? We use "select" a lot in our docs, but here the user isn't selecting anything—just clicking or tapping inside a space.

@mushtaqak I'd rather keep "this space" instead of "here"—it's more specific and I think will help avoid confusion.

roderickmorales commented:
@srpearce I added a screenshot of what github does above

here is what dropbox does:
![screen shot 2017-08-08 at 4 58 41 pm](https://user-images.githubusercontent.com/1260455/29094329-fda95bba-7c5a-11e7-8b4e-1585b1fde8fc.png)

this is what JIRA does:
![screen shot 2017-08-08 at 5 00 47 pm](https://user-images.githubusercontent.com/1260455/29094388-2a86b72c-7c5b-11e7-8ef8-68f1b704e4bc.png)

We should try to move away from "click" every chance we get. Here is a very helpful article that articulates very well why not to use "click" or "here"

muzaffaryousaf commented:
@srpearce @roderickmorales I think the string should be this `Drag and drop or browse your computer`. I've already discuss this with Roderick. Does this sounds good to you guys ?

srpearce commented:
@muzaffaryousaf @roderickmorales That's fine with me.

muzaffaryousaf commented:
@muhammad-ammar can you make this change ^^ and merge the PR before we start next sprint ?
